app/models/service_contract.rb

```diff
@@ -31,6 +31,10 @@ class ServiceContract < Contract
     where(:plans => { :issuer_id => service_id.to_i }).joins(:plan).references(:plan)
   end
 
+  # TODO: unit test this scope
+  def self.provided_by(account)
+    where(plan_id: ServicePlan.provided_by(account).select(:id))
+  end
```

Comment:
This looks like a scope. Maybe define it as a such instead of a class method? Something like:
```suggestion
  scope :provided_by, -> (account) do
    where(plan_id: ServicePlan.provided_by(account).select(:id))
  end
```